src/actions/index.js

```diff
@@ -0,0 +1,53 @@
+import bookService from '../services/books'
+
+export async function fetchBooks (dispatch) {
+  dispatch({ type: 'START_LOADER' })
+  try {
+    const books = await bookService.getAll()
+    dispatch({
+      type: 'SET_BOOKS_COLLECTION',
+      books
+    })
+  } catch (error) {
+    console.log(error)
+  } finally {
+    dispatch({ type: 'STOP_LOADER' })
+  }
+}
+
+export async function createBook (dispatch, book) {
```

Comment:
What do you think about this:

```javascript
export async function createBook(book){
  return function(dispatch){
   // same things you put inside 
 }
}
```

This is the more proper way of writing a "thunk" action.

Now of course this may mean we need to have a middleware to handle an action that returns an action. There is a very light-weight middleware to handle "thunks" -> `redux-thunk` to help us with this. It makes our actions much cleaner to use in the connected components later on (see my comment on `App.js`)